internal/util/searchreplace/searchreplace.go

```diff
@@ -0,0 +1,174 @@
+package searchreplace
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt/pkg/kptfile"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const PathDelimiter = "."
+
+// SearchReplace struct holds the input parameters and results for
+// Search and Replace operations on resource configs
+type SearchReplace struct {
+	// Value is the value of the field to be matched
+	Value string
+
+	// ValueRegex is the value regex of the field to be matched
+	ValueRegex string
+
+	// Path is the path of the field to be matched
+	Path string
+
+	// Count is the number of matches
+	Count int
+
+	// Match is the map of file path to list of matched fields
+	Match map[string][]string
+
+	// PutLiteral is the literal to be put at to field
+	// filtered by path and/or value
+	PutLiteral string
+
+	filePath string
+}
+
+// Perform performs the search and replace operation on each node in the package path
+func (sr *SearchReplace) Perform(resourcesPath string) error {
+	inout := &kio.LocalPackageReadWriter{
+		PackagePath:     resourcesPath,
+		NoDeleteFiles:   true,
+		PackageFileName: kptfile.KptFileName,
+	}
+	sr.Match = make(map[string][]string)
+	return kio.Pipeline{
+		Inputs:  []kio.Reader{inout},
+		Filters: []kio.Filter{kio.FilterAll(sr)},
+		Outputs: []kio.Writer{inout},
+	}.Execute()
+}
+
+// Filter parses input node and performs search and replace operation on the node
+func (sr *SearchReplace) Filter(object *yaml.RNode) (*yaml.RNode, error) {
+	filePath, _, err := kioutil.GetFileAnnotations(object)
+	if err != nil {
+		return object, err
+	}
+	sr.filePath = filePath
+
+	if isAbsPath(sr.Path) && sr.Value == "" && sr.ValueRegex == "" && sr.PutLiteral != "" {
```

Comment:
Maybe put this check into a function so we can give it a name. Should make it easier to understand what it is doing.